openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -654,34 +659,40 @@ def get_queryset(self) -> TagDataQuerySet:
         taxonomy_id = int(self.kwargs.get("pk"))
         taxonomy = self.get_taxonomy(taxonomy_id)
         parent_tag_value = self.request.query_params.get("parent_tag", None)
-        root_only = "root_only" in self.request.query_params
         include_counts = "include_counts" in self.request.query_params
         search_term = self.request.query_params.get("search_term", None)
-
-        if parent_tag_value:
-            # Fetching tags below a certain parent is always paginated and only returns the direct children
-            depth = 1
-            if root_only:
-                raise ValidationError("?root_only and ?parent_tag cannot be used together")
+        full_depth_threshold = int(self.request.query_params.get("full_depth_threshold", 0))
+        if full_depth_threshold < 0 or full_depth_threshold > MAX_FULL_DEPTH_THRESHOLD:
+            raise ValidationError("Invalid full_depth_threshold")
```

Comment:
Can we handle invalid numbers here too?

```suggestion
        try:
            full_depth_threshold = int(self.request.query_params.get("full_depth_threshold", 0))
        except ValueError:
            full_depth_threshold = -1
        if full_depth_threshold < 0 or full_depth_threshold > MAX_FULL_DEPTH_THRESHOLD:
            raise ValidationError("Invalid full_depth_threshold")
```